### PR TITLE
Python 3.6.0

### PR DESCRIPTION
**Pull request by jjhelmus**

### CODE REVIEW

**Comment by msarahan** [comment]
Weird default on Travis. Restarted.

**Comment by jjhelmus** [comment]
Incorporated changes in #97 and #87 into this update.  @xmnlab and @pitrou can you review these changes and see if they meet the suggestions you made?

**Comment by jjhelmus** [comment]
Is there any interest in updating the pinning to more recent versions with this new version of Python?  Specifically the following could be updated:

* readline `6.2*` -> `6.3*`
* sqlite `3.13.*` -> `3.15.*`
* tk `8.5.*` -> `8.6.*`

The version in the `defaults` channel does not seem to change the pinning.

**Comment by jjhelmus** [comment]
Reverting the changes in #97 as these changes seem to cause [issues with the Linux build](https://circleci.com/gh/jjhelmus/python-feedstock/3?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

**Comment by jjhelmus** [comment]
> Weird segfault on Travis.

This seems to be real, I can replicate this locally as well.  Tried a few changes and cannot figure out what is the root cause.  I see the same issue with the recipe in #100.

**Comment by jakirkham** [comment]
If we update the pinnings, then they deviate from the pinnings that we are using with everything else.

**Comment by msarahan** [comment]
@jjhelmus the segfault had something to do with conda build 2?

If so, it is probably the prefix replacement or rpath stuff messing things up.

**Comment by scopatz** [comment]
This looks good to me FWIW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
       - omit-local-site-packages.patch  # [osx]
       - osx64-dist.patch                # [osx and x86_64]
       - win-find_exe.patch              # [win]
       - win-library_bin.patch           # [win]
+      # Adjust FD_SETSIZE to allow select() to wait on more connections
+      # simultaneously.  Desirable for serious use of Tornado on Windows.
+      # See also:
+      # - https://bugs.python.org/issue28708
+      # - https://github.com/zeromq/libzmq/issues/1165
+      - fd_setsize.patch                # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Maybe we should be taking this on separately. Namely as a PR ( https://github.com/conda-forge/python-feedstock/pull/87 ). Doesn't feel like the sort of thing we should bury in a release.

**Comment by pitrou** [comment]
Why isn't the feedstock versioned? The way this is done, it seems to suggest that conda-forge won't provide bugfixes for the Python 3.5 package anymore.

**Comment by jakirkham** [comment]
There's already a `3.5` branch. `master` always contains the latest version.

That said, we may decide to stop supporting some versions/branches like `3.4`, but I expect we will be supporting `3.5` for a while.

**Comment by pitrou** [comment]
I see. Thanks for correcting me.

**Comment by ocefpaf** [comment]
@jjhelmus what is the status of this? Can we merge it?

**Comment by jjhelmus** [comment]
> @jjhelmus what is the status of this? Can we merge it?

I think there is an issue with the OS X build when compiling with a long prefix environment.  Let me try to debug the issue in the next day or two.

**Comment by jjhelmus** [comment]
No OS X issue!  I think this may be good to go if others concur.  

I can pull the fd_size patch out an submit is as a separate PR if that is preferred.

File: recipe/build.sh [hunk 2]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 
-python ${RECIPE_DIR}/brand_python.py
+${SYS_PYTHON} ${RECIPE_DIR}/brand_python.py
+
+# Remove test data and ensurepip stubs to save space
+rm -rf Lib/test Lib/*/test
+rm -rf Lib/ensurepip
```
Comment (jjhelmus) [inline recipe/build.sh]: Removing these files saves a significant amount of space, perhaps the same should be done in the Window build.

Comment (scopatz) [inline recipe/build.sh]: +1, but maybe that can be done in a separate  pr

Comment (jjhelmus) [inline recipe/build.sh]: I made the needed changes. @Nelthorim (thanks!) had them in his Python 3.6 PR so they were not hard to find.

**Comment by jjhelmus** [comment]
I think this should be good to go now.  Can someone merge this or give a :+1: and I'll merge?

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipe/meta.yaml]: Any thoughts on this point, @jjhelmus?

Comment (jjhelmus) [inline recipe/meta.yaml]: I think it can be left it as the change is fine, but would be happy to pull the patch out and submit as a separate PR.

**Comment by scopatz** [comment]
Woohoo!
